### PR TITLE
adding tests for cluster-samples-operator repository for cluster with no-capabilities

### PR DESCRIPTION
**Pull request by tisutisu**
Trying to add samples-operator specific checks when samples operator is not present in the cluster.
This PR will address story: https://issues.redhat.com/browse/SO-56

### CODE REVIEW

File: ci-operator/config/openshift/cluster-samples-operator/openshift-cluster-samples-operator-release-4.11.yaml [hunk 1]
```diff
@@ -51,6 +51,22 @@ tests:
...
+  skip_if_only_changed: ^docs/|\.md$|^(?:.*/)?(?:\.gitignore|OWNERS|PROJECT|LICENSE)$
+  steps:
+    cluster_profile: aws
+    env:
+      BASELINE_CAPABILITY_SET: None
+    test:
+    - as: test
+      commands: |
+        oc get templates -n openshift | grep 'No resources found'
+        oc get is -n openshift | grep 'No resources found'
```
[the remark this answers is not in the document]
Comment (tisutisu) [reply ci-operator/config/openshift/cluster-samples-operator/openshift-cluster-samples-operator-release-4.11.yaml]: Thanks for pointing it out. Added this change in the latest commit 2b34072

File: ci-operator/config/openshift/cluster-samples-operator/openshift-cluster-samples-operator-release-4.11.yaml [hunk 2]
```diff
@@ -51,6 +51,29 @@ tests:
...
+  steps:
+    cluster_profile: aws
+    env:
+      BASELINE_CAPABILITY_SET: None
+    test:
+    - as: test
+      commands: |
+        function assert_empty() {
+          KIND="$1"
+          NAMES="$(oc -n openshift get -o jsonpath='{.items[*].metadata.name}' "${KIND}")"
```
Comment (wking) [inline ci-operator/config/openshift/cluster-samples-operator/openshift-cluster-samples-operator-release-4.11.yaml:65]: [rehearsal][1]:

```
INFO[2022-06-02T11:10:21Z] environment: line 4: oc: command not found
```

Take a look at [these docs][2] for addressing that.

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/29048/rehearse-29048-pull-ci-openshift-cluster-samples-operator-release-4.11-check-aws-no-capabilities/1532307034429460480#1:build-log.txt%3A42
[2]: https://docs.ci.openshift.org/docs/architecture/step-registry/#injecting-the-oc-cli

Comment (tisutisu) [reply ci-operator/config/openshift/cluster-samples-operator/openshift-cluster-samples-operator-release-4.11.yaml:66]: Addressed this issue in the latest commit 44048e9

File: ci-operator/config/openshift/cluster-samples-operator/openshift-cluster-samples-operator-release-4.11.yaml [hunk 3]
```diff
@@ -51,6 +51,30 @@ tests:
...
+    test:
+    - as: test
+      cli: latest
+      commands: |
+        function assert_empty() {
+          KIND="$1"
+          NAMES="$(oc -n openshift get -o jsonpath='{.items[*].metadata.name}' "${KIND}")"
+          if test -n "${NAMES}"
+          then
+            echo "Should not have ${KIND} in the openshift namespace, but we have: ${NAMES}"
```
[the remark this answers is not in the document]
Comment (tisutisu) [reply ci-operator/config/openshift/cluster-samples-operator/openshift-cluster-samples-operator-release-4.11.yaml]: Addressed this issue in the latest commit cf9110c . @wking Could you please take a look again?

File: ci-operator/config/openshift/cluster-samples-operator/openshift-cluster-samples-operator-release-4.11.yaml [hunk 4]
```diff
@@ -51,6 +51,58 @@ tests:
...
+            fi
+          done
+          return 1
+        }
+
+        function check_present_list_is_correct() {
+          CURRENT_LIST=("$@")
+          for i in "${CURRENT_LIST[@]}";
+          do
+            if exists_in_list "$SHOULD_BE_PRESENT" " " $i; then
```
[the remark this answers is not in the document]
Comment (tisutisu) [reply ci-operator/config/openshift/cluster-samples-operator/openshift-cluster-samples-operator-release-4.11.yaml]: @wking Thanks for pointing it out. separated allowed list for templates and imagestreams in the latest commit 1d2f2a5 . Hope with this change will address the issue mentioned.

File: ci-operator/config/openshift/cluster-samples-operator/openshift-cluster-samples-operator-release-4.11.yaml [hunk 5]
```diff
@@ -51,6 +51,74 @@ tests:
...
+              else
+                echo "Unknown kind passed as an argument"
+                return 1
+              fi
+            done
+        }
+
+        function assert_empty() {
+          KIND="$1"
+          NAMES="$(oc -n openshift get -o jsonpath='{.items[*].metadata.name}' "${KIND}")"
```
Comment (wking) [inline ci-operator/config/openshift/cluster-samples-operator/openshift-cluster-samples-operator-release-4.11.yaml:109]: If you want `NAMES` to be a Bash array, so `"${NAMES[@]}"` will expand to multiple arguments, you should use:

```sh
NAMES=($(oc -n openshift get -o jsonpath='{.items[*].metadata.name}' "${KIND}"))
```

Comment (tisutisu) [reply ci-operator/config/openshift/cluster-samples-operator/openshift-cluster-samples-operator-release-4.11.yaml]: changed name variable to be a Bash array and did the subsequent changes as well. @wking could you please review once again?

File: ci-operator/config/openshift/cluster-samples-operator/openshift-cluster-samples-operator-release-4.11.yaml [hunk 6]
```diff
@@ -51,6 +51,74 @@ tests:
...
+            fi
+          done
+          return 1
+        }
+
+        function check_present_list_is_correct() {
+          CURRENT_LIST=("$@")
+          ((last_idx=${#CURRENT_LIST[@]} - 1))
+          KIND=${CURRENT_LIST[last_idx]}
+          unset CURRENT_LIST[last_idx]
```
[the remark this answers is not in the document]
Comment (tisutisu) [reply ci-operator/config/openshift/cluster-samples-operator/openshift-cluster-samples-operator-release-4.11.yaml]: Added this change.

**Review by dperaza4dustbit — APPROVED** [review]
/lgtm

File: ci-operator/config/openshift/cluster-samples-operator/openshift-cluster-samples-operator-release-4.11.yaml [hunk 7]
```diff
@@ -51,6 +51,72 @@ tests:
...
+            do
+              if [ "$KIND" == "imagestreams" ] ; then
+                if exists_in_list "$IMAGESTREAM_ALLOWED_LIST" " " $i; then
+                  echo "imagestream $i is present, which is expected"
+                else
+                  echo "$i imagestream should not be present"
+                  return 1
+                fi
+              elif [ "$KIND" == "templates" ] ; then
+                if exists_in_list "$TEMPLATE_ALLOWED_LIST" " " $i; then
```
Comment (wking) [inline ci-operator/config/openshift/cluster-samples-operator/openshift-cluster-samples-operator-release-4.11.yaml:93]: You can avoid this switching with:

```sh
assert_empty templates "${TEMPLATE_ALLOWED_LIST}" &&
assert_empty imagestreams "${IMAGESTREAM_ALLOWED_LIST}"
```

But :shrug:, it should work fine this way too.

/lgtm

**Pull request merged**